contracts/contracts/strategies/BaseConvexMetaStrategy.sol

```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity ^0.8.0;
+
+/**
+ * @title Curve Convex Strategy
+ * @notice Investment strategy for investing stablecoins via Curve 3Pool
+ * @author Origin Protocol Inc
+ */
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import { IRewardStaking } from "./IRewardStaking.sol";
+import { ICurvePool } from "./ICurvePool.sol";
+import { ICurveMetaPool } from "./ICurveMetaPool.sol";
+import { IERC20, BaseCurveStrategy } from "./BaseCurveStrategy.sol";
+import { StableMath } from "../utils/StableMath.sol";
+import { Helpers } from "../utils/Helpers.sol";
+
+abstract contract BaseConvexMetaStrategy is BaseCurveStrategy {
+    using StableMath for uint256;
+    using SafeERC20 for IERC20;
+    event MaxWithdrawalSlippageUpdated(
+        uint256 _prevMaxSlippagePercentage,
+        uint256 _newMaxSlippagePercentage
+    );
+
+    // used to circumvent the stack too deep issue
+    struct InitState {
```

Comment:
`InitConfig` is probably a better name for this than `InitState`.

Reply:
agreed, fix: https://github.com/OriginProtocol/origin-dollar/pull/1002/commits/b6b5a3b00d524a651ba517c5f1cb261552607456

Reply:
Fix confirmed. Comment resolved.